From d85e8240bb8f92f057b56c31de9f708f5891fe7c Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Thu, 18 Apr 2024 16:16:13 +0300
Subject: [PATCH] [SW-15774] build: fs1: move hl_pldm_hex

Make hl_pldm_hex a reusable function by moving it to fs1-image-utility.sh

Change-Id: Ieaf3cec01d9a6ed70fa9d01a4f56b87c7a5b533e
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/fs1_arc/fs1-image-utility.sh b/board/habanalabs/fs1_arc/fs1-image-utility.sh
index 2bc2530..1497054 100755
--- a/board/habanalabs/fs1_arc/fs1-image-utility.sh
+++ b/board/habanalabs/fs1_arc/fs1-image-utility.sh
@@ -191,3 +191,22 @@
 	# Clean up
 	rm -f tmp
 }
+
+function hl_pldm_hex {
+	local NAME=$(echo "$1" | cut -f 1 -d '.')
+	local FILE=$2
+	local BYTES_PER_LINE=$3
+	local DESTDIR=$4
+	# pad file to 128bit (16Byte) as pladium emulates 128bit memory, so xxd wont trunc lines
+	dd status=none conv=sync bs=16 if=${FILE} > ${FILE}.tmp
+	xxd -e -g ${BYTES_PER_LINE} ${FILE}.tmp | xxd -r | xxd -ps -c${BYTES_PER_LINE} > ${DESTDIR}/${DEV_PREFIX}-${HL_TARGET}-${NAME}.hex
+	rm ${FILE}.tmp
+}
+
+function hl_pldm_flash_hex {
+	local NAME=$(echo "$1" | cut -f 1 -d '.')
+	local FILE=$2
+	local DESTDIR=$3
+	# save hex as a 1 byte list of values
+	xxd -e -g 4 ${FILE} | xxd -r | xxd -ps -c1 > ${DESTDIR}/${DEV_PREFIX}-${HL_TARGET}-${NAME}.hex
+}
diff --git a/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh b/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
index 558ca9c..8cc5831 100755
--- a/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
+++ b/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
@@ -10,7 +10,7 @@
 BOARD_BASE=${HL_BASE_PATH}/${HL_DEV}
 DEVICE_BASE=${HL_BASE_PATH}/${HL_DEV}
 KEY_BASE=${BR2_EXTERNAL}/board/habanalabs/${DEV_PREFIX}_arc
-OUTPUT_DIR=${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/images
+source ${DEVICE_BASE}/fs1-image-utility.sh
 OBJCOPY=objcopy
 ARCBOOT_BINFILE=${ARCBOOT_BINFILE:-boot_rom.bin}
 PPBOOT_BINFILE=${PPBOOT_BINFILE:-ppboot.bin}
@@ -79,23 +79,6 @@
 	hl_reverse_words ${FILE} ${FILE}.out 4
 	CRC32=$(crc32 ${FILE}.out)
 	echo ${CRC32} | xxd -r -p > ${FILE}.crc32
-}
-
-function hl_pldm_hex {
-	NAME=$(echo "$1" | cut -f 1 -d '.')
-	FILE=$2
-	BYTES_PER_LINE=$3
-	# pad file to 128bit (16Byte) as pladium emulates 128bit memory, so xxd wont trunc lines
-	dd status=none conv=sync bs=16 if=${FILE} > ${FILE}.tmp
-	xxd -e -g ${BYTES_PER_LINE} ${FILE}.tmp | xxd -r | xxd -ps -c${BYTES_PER_LINE} > ${DEV_PREFIX}-${HL_TARGET}-${NAME}.hex
-	rm ${FILE}.tmp
-}
-
-function hl_pldm_flash_hex {
-	NAME=$(echo "$1" | cut -f 1 -d '.')
-	FILE=$2
-	# save hex as a 1 byte list of values
-	xxd -e -g 4 ${FILE} | xxd -r | xxd -ps -c1			> ${DEV_PREFIX}-${HL_TARGET}-${NAME}.hex
 }
 
 function hl_sec_fw {
